src/components/gitOps/GitOpsConfiguration.tsx

```diff
@@ -190,267 +201,536 @@ class GitOpsConfiguration extends Component<GitOpsProps, GitOpsState> {
-                        onChange={(event) => { this.handleChange(event, key); }}
-                        labelClassName="gitops__id form__label--fs-13 fw-5 fs-13" />
-                </div>
-                {this.state.providerTab === GitProvider.BITBUCKET_CLOUD &&
-                <div className="mt-4 flex left">
-                   <InfoFill className="icon-dim-16" />
-                   <span className="ml-4 fs-11">If the project is not provided, the repository is automatically assigned to the oldest project in the workspace.</span>
-                </div>
-                }
-                <hr />
-                <div className="fw-6 cn-9 fs-14 mb-16">Git access credentials</div>
-
-                <div className="form__row--two-third gitops__id mb-20 fs-13">
-                    <div>
-                        <CustomInput autoComplete="off"
-                            value={this.state.form.username}
-                            onChange={(event) => this.handleChange(event, 'username')}
-                            name="Enter username"
-                            error={this.state.isError.username}
-                            tabIndex={3}
-                            label={this.state.providerTab === GitProvider.GITLAB ? "GitLab Username*" : this.state.providerTab === GitProvider.AZURE_DEVOPS ? "Azure DevOps Username*" : this.state.providerTab === GitProvider.BITBUCKET_CLOUD ? "Bitbucket Username*" : "GitHub Username*"}
-                            labelClassName="gitops__id form__label--fs-13 fw-5 fs-13" />
+                            link={
+                                this.state.providerTab === GitProvider.GITLAB
+                                    ? GitLink.GITLAB
+                                    : this.state.providerTab === GitProvider.AZURE_DEVOPS
+                                    ? GitLink.AZURE_DEVOPS
+                                    : this.state.providerTab === GitProvider.BITBUCKET_CLOUD
+                                    ? GitLink.BITBUCKET_PROJECT
+                                    : GitLink.GITHUB
+                            }
+                            linkText={
+                                this.state.providerTab === GitProvider.GITLAB
+                                    ? '(How to create group in GitLab?)'
+                                    : this.state.providerTab === GitProvider.AZURE_DEVOPS
+                                    ? '(How to create project in Azure?)'
+                                    : this.state.providerTab === GitProvider.BITBUCKET_CLOUD
+                                    ? '(How to create project in bitbucket?)'
+                                    : '(How to create organization in GitHub?)'
+                            }
+                            label={
```

Comment:
too many times logic has been put in the template, cant we derive it with state? or using just Map instead

Reply:
same for other places in this code where conditional check is made

Reply:
done, have used maps for this